### PR TITLE
Add BCE when no annotations

### PR DESCRIPTION
**Pull request by rvandeghen**
Wrong implementaion of focal loss ?

### CODE REVIEW

File: retinanet/losses.py [hunk 1]
```diff
@@ -48,12 +48,35 @@ def forward(self, classifications, regressions, anchors, annotations):
...
+
+                    alpha_factor = 1. - alpha_factor
+                    focal_weight = classification
+                    focal_weight = alpha_factor * torch.pow(focal_weight, gamma)
+
+                    bce = -((1.0 - targets) * torch.log(1.0 - classification))
+
+                    # cls_loss = focal_weight * torch.pow(bce, gamma)
+                    cls_loss = focal_weight * bce
+                    classification_losses.append(cls_loss.sum(), min=1.0))
```
Comment (zylo117) [inline retinanet/losses.py]: missing clamp?

Comment (rvandeghen) [reply retinanet/losses.py]: changed

File: retinanet/losses.py [hunk 2]
```diff
@@ -45,18 +45,41 @@ def forward(self, classifications, regressions, anchors, annotations):
...
+
+                    alpha_factor = 1. - alpha_factor
+                    focal_weight = classification
+                    focal_weight = alpha_factor * torch.pow(focal_weight, gamma)
+
+                    bce = -((1.0 - targets) * torch.log(1.0 - classification))
+
+                    # cls_loss = focal_weight * torch.pow(bce, gamma)
+                    cls_loss = focal_weight * bce
+                    classification_losses.append(cls_loss.sum(), min=1.0)
```
Comment (zylo117) [inline retinanet/losses.py:64]: still here, it should be `classification_losses.append(cls_loss.sum())`, right?

Comment (rvandeghen) [reply retinanet/losses.py:64]: Yep

**Comment by yhenon** [comment]
This is awesome, thanks.
